federatedscope/core/data/base_translator.py

```diff
@@ -0,0 +1,123 @@
+import logging
+import numpy as np
+
+from federatedscope.core.auxiliaries.splitter_builder import get_splitter
+from federatedscope.core.data import ClientData, StandaloneDataDict
+
+logger = logging.getLogger(__name__)
+
+
+class BaseDataTranslator:
+    """
+    Perform process:
+        Dataset -> ML split -> FL split -> Data (passed to FedRunner)
+
+    """
+    def __init__(self, global_cfg, client_cfgs=None):
+        """
+        Convert data to `StandaloneDataDict`.
+
+        Args:
+            global_cfg: global CfgNode
+            client_cfgs: client cfg `Dict`
+        """
+        self.global_cfg = global_cfg.clone()
```

Comment:
In passing cfg object(s), when do we clone the object? It seems that, in our repository, there is no a unified manner.

Reply:
We clone `cfg` when it will potentially be modified.